### PR TITLE
Add filter functionality back to library PAY-1771

### PR DESCRIPTION
**Pull request by nicoback2**
- Make sure tracks/collections filter query works on library tabs
- Note: There is no filter feature on web library collections tabs

Screenshot 2023-09-06 at 5 03 00 PM

### CODE REVIEW

File: packages/common/src/utils/collectionUtils.ts [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+export function filterCollections<
+  T extends AccountCollection | EnhancedCollection
+>(collections: T[], { filterText = '' }: FilterCollectionsOptions): T[] {
+  return collections.filter((item: AccountCollection | EnhancedCollection) => {
+    const matchesOwnerName =
+      item.user.handle.toLowerCase().indexOf(filterText.toLowerCase()) > -1
+    let matchesPlaylistName: boolean
+
+    if (isAccountCollection(item)) {
+      matchesPlaylistName =
```
Comment (dylanjeffers) [inline packages/common/src/utils/collectionUtils.ts:23]: could possibly just do `const name = isAccountCollection(item) ? item.name : item.playlist_name` and then do the filter on that.

File: packages/mobile/src/screens/favorites-screen/useCollectionsScreenData.ts [hunk 2]
```diff
@@ -87,8 +91,16 @@ export const useCollectionsScreenData = ({
   const availableCollectionIds = useProxySelector(
     (state: AppState) => {
       if (isReachable) {
-        return [...locallyAddedCollections, ...fetchedCollectionIds].filter(
-          (id) => !locallyRemovedCollections.has(id)
+        const filteredLocallyAddedCollections = filterCollections(
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/favorites-screen/useCollectionsScreenData.ts:94]: might want to add the Ids postfix?

**Review by dylanjeffers — APPROVED** [review]
great stuff

File: packages/common/src/utils/collectionUtils.ts (hunk 1 above)
Comment (nicoback2) [reply packages/common/src/utils/collectionUtils.ts:23]: ah yess

File: packages/mobile/src/screens/favorites-screen/useCollectionsScreenData.ts (hunk 2 above)
Comment (nicoback2) [reply packages/mobile/src/screens/favorites-screen/useCollectionsScreenData.ts:94]: ah sorry wdym?

Comment (nicoback2) [reply packages/mobile/src/screens/favorites-screen/useCollectionsScreenData.ts:94]: ohhh yea i get it, will do

Comment (dylanjeffers) [reply packages/mobile/src/screens/favorites-screen/useCollectionsScreenData.ts:94]: Sorry yeah :)

**Pull request merged**